pageserver/src/tenant_tasks.rs

```diff
@@ -290,22 +290,32 @@ async fn wait_for_active_tenant(
         }
     };
 
-    if tenant.current_state() != TenantState::Active {
+    if !matches!(
+        tenant.current_state(),
+        TenantState::Active {
+            background_jobs_running: true
+        }
+    ) {
         let mut tenant_state_updates = tenant.get_state_updates_watch();
```

Comment:
`!matches!` looks exotic, took me a few seconds to parse that in my head :-). How about

```
if tenant.current_state() != TenantState::Active { background_jobs_running: true }
```
?

Reply:
Or maybe remove the if-check altogether and go straight into the loop:
```
loop {
    let new_state = *tenant_state_updates.borrow();
    match new_state { 
        ...
    }
    if Err(_sender_dropped_error) = tenant_state_updates.changed().await {
        info!("Tenant dropped the state updates sender, quitting waiting for tenant and the task loop");
        return ControlFlow::Break(());
    }
}